### PR TITLE
Pass required flags to controller manager so that pods CIDR can be custom

### PR DESCRIPTION
**Pull request by fiunchinho**
### What this PR does / why we need it

Towards https://github.com/giantswarm/roadmap/issues/1347

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create

### CODE REVIEW

**Review by AverageMarcus — APPROVED** [review]

**Review by whites11 — APPROVED** [review]

**Comment by fiunchinho** [comment]
It failed with 

```
failed to create resource: admission webhook
      "default.kubeadmcontrolplane.controlplane.cluster.x-k8s.io" denied the request:
      json: cannot unmarshal array into Go struct field ControlPlaneComponent.spec.kubeadmConfigSpec.clusterConfiguration.controllerManager.extraArgs
      of type string'
```

I changed the code to take the first element of the CIDRs list. Please take a look.

File: helm/cluster-gcp/templates/_control_plane.tpl [hunk 1]
```diff
@@ -69,11 +69,12 @@ spec:
           pathType: DirectoryOrCreate
       controllerManager:
         extraArgs:
-          allocate-node-cidrs: "false"
+          allocate-node-cidrs: "true"
           authorization-always-allow-paths: "/healthz,/readyz,/livez,/metrics"
           bind-address: 0.0.0.0
           cloud-provider: gce
           cloud-config: /etc/kubernetes/gcp.conf
+          cluster-cidr: {{ .Values.network.podCidr | first }}
```
[the remark this answers is not in the document]
Comment (whites11) [reply helm/cluster-gcp/templates/_control_plane.tpl:77]: Comma separated

https://github.com/kubernetes/kubernetes/blob/master/cmd/cloud-controller-manager/nodeipamcontroller.go#L154

[the remark this answers is not in the document]
Comment (fiunchinho) [reply helm/cluster-gcp/templates/_control_plane.tpl:77]: Thanks Chris :heart:

[the remark this answers is not in the document]
Comment (whites11) [reply helm/cluster-gcp/templates/_control_plane.tpl:77]: It should be

```
join "," .Values.network.podCidr
```

**Review by whites11 — APPROVED** [review]

**Pull request merged**